platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,852 @@
+    random.nextBytes(seed);
+    final FieldElement anotherElement = field.randomElement(seed);
+    final FieldElement resultAddition = randomElement.add(anotherElement);
+    final FieldElement resultMultiplication = randomElement.multiply(anotherElement);
+}
+```
+###### Work with the groups associated to the pairing
+```java
+static {
+    final Group group1 = pairing.getGroup1();
+    final Group group2 = pairing.getGroup2();
+
+    final Group g2 = group1.getOppositeGroup();
+    final GroupElement g1Generator = group1.getGenerator();
+    final GroupElement g2Generator = group2.getGenerator();
+    final GroupElement zero = group1.zeroElement();
+
+    final byte[] serialization = zero.toBytes();
+    
+    final GroupElement element =  group1.elementFromBytes(/*Some byte array representing a group element*/);
+    g1Generator.isSameGroup(g2Generator); //false
+}
+```
+
+
+#### Swirlds Crypto Pairings Signature Library
+##### Overview
+This module provides cryptography primitives to create EC PublicKeys, EC PrivateKeys, and Signatures.
+
+##### Public API
+###### `SignatureSchema`
+**Link**: [SignatureSchema.java](signature-lib%2FSignatureSchema.java)
+
+**Description**: A pairings signature scheme can be implemented with different types of curves and group assignment configurations.
+For example, two different configurations might consist of a BLS_12_381 curve using G1 of the pairing to generate public key elements or G2 for the same purpose.
+###### `PairingPrivateKey`
+**Link**: [PairingPrivateKey.java](signature-lib%2FPairingPrivateKey.java)
+
+**Description**: A private key generated using the pairings API
+
+###### `PairingPublicKey`
+**Link**: [PairingPublicKey.java](signature-lib%2FPairingPublicKey.java)
+
+**Description**: A public key generated using the pairings API
+###### `PairingSignature`
+**Link**: [PairingSignature.java](signature-lib%2FPairingSignature.java)
+
+**Description**: A signature generated with the private key that can be verified with the public key
+
+###### `GroupAssignment`
+**Link**: [GroupAssignment.java](signature-lib%2FGroupAssignment.java)
+
+**Description**: An enum to clarify which group public keys and signatures are in, for a given SignatureSchema
+
+
+##### Implementation Note
+The serialization of the elements in this module adds a byte to represent the combination of Curve type and group assignment.
+
+##### Examples
+###### Generating a SignatureSchema
+```java
+SignatureSchema signatureSchema = SignatureSchema.from(Curve.ALT_BN128, GroupAssignament.G1_PUBLIC_KEYS);
+```
+###### Generating a Private Key
+```java
+   PairingPrivateKey privateKey = PairingPrivateKey.create(signatureSchema, new SecureRandom());
+   
+```
+###### Generating a Public Key
+```java
+   PairingPublicKey publicKey = PairingPublicKey.create(privateKey);
+   
+```
+###### Generating a Signature
+```java
+    byte[] message = new byte[]{};
+    PairingSignature signature = PairingSignature.sign(privateKey, message );
```

Comment:
I believe the modification below is correct, since it matches the java file provided. I consider this change immaterial since the actual java API provided is correct.
```suggestion
    PairingSignature signature = PairingPrivateKey.sign(message );
```